rust/origen/src/testers/v93k/mod.rs

```diff
@@ -46,6 +46,21 @@ impl Renderer {
         }
         Ok(Return::Unmodified)
     }
+
+    fn char_mapper(&self, action: PinActions, data: u8) -> char {
+        match action {
+            PinActions::Drive => match data {
+                0 => '0',
+                _ => '1',
+            },
+            PinActions::Verify => match data {
+                0 => 'L',
+                _ => 'H',
+            },
+            PinActions::Capture => 'C',
+            PinActions::HighZ => 'Z',
+        }
```

Comment:
I think this might be a bit over-simplified by mapping an action to a character and not considering pin and timeset specific mappings.
i.e. on the V93K the character is defined by the timeset and the pin - e.g. it's common in V93K patterns to use 'P' for drive 1 on clock pins.

I can't point you to where such things are modeled without digging in further, but the same concepts are possible in STIL and @coreyeng setup the timeset model to be STIL-compatible.

It's fine to start with a simplified approach of course, but I think in that case I would like to at least see a pin reference coming in here (and the timeset if we can't look it up from here), even if we don't actually use it yet.

Reply:
I wasn't attempting to implement a 93K mapping. I just duplicated the same mapping that was already being done by pins as an example of how to use the feature. Somebody who knows the 93K would need to implement the actual mapping.

The pin (or group if it was a group) name is available from the iterator. I don't think the individual pin names are passed all the way through. I'll look into how to also pass through the individual pin name, or it should be easy to figure out the pin by passing the group name and the enumerator for the pin in the group.

This Renderer tracks the current timeset as it processes the AST and stores it as an attribute. That's probably the way to go for that.

Reply:
For reference this is what was being use to do the mapping

https://github.com/Origen-SDK/o2/blob/e49e5df6e305f7ce04e2be6ceeffbb833ae2113a/rust/origen/src/core/model/pins/pin.rs#L51-L65

Reply:
added pin name to data that gets yielded

Reply:
I think the 93K could use what's here to further mess with the characters. The V93K will have the timeset and I think it'd be the 93K's responsibility to figure out what the mapping is. Obviously I didn't put much effort into these aspects on the first go-around, so I'm fine leaving that as a future item to get IGXL testers going.

I still think there's work to do in extracting out the common elements of rendering and getting as much semblance as possible in the renderer implementations. I'd like to see the `char_mapper` as a trait function on the `Renderer`, that could even default to just `to_string`, though that's probably not necessary just to get the ball rolling on IGXL generation.